## Batch: H
## Group name: H1
## Group member name: Hebbar Sumukh

In [4]:
import os
os.chdir('C:/Users/nipanda/Documents/Capstone Project1_Marketing Data/Capstone Project1_Marketing Data/Marketing_Campaign_P1')

# 1.Analysis of percentage turnout of marketing campaign

In [5]:
import pandas as pd
import numpy as np

In [6]:
turnout=pd.read_csv('Customer_Response_data_p1.csv')
turnout.head()

Customer_id   y
0            1  no
1            2  no
2            3  no
3            4  no
4            5  no

In [7]:
no,yes=turnout['y'].value_counts()
print('Total number with no response:',no)
print('Total number with yes response:',yes)


Total number with no response: 32876
Total number with yes response: 4208


In [8]:
total=turnout['y'].count()
print('Total Customers:',total)

Total Customers: 37084


In [9]:
pt=(yes*100)/total
print('Percentage Turnout(the percentage of people optiong for term deposit):',pt)

Percentage Turnout(the percentage of people optiong for term deposit): 11.34721173551936


# 2.	Right mode to contact the customers (telephone or mobile)

In [1]:
import pandas as pd
import numpy as np

In [6]:
city=pd.read_csv("City_Master.csv")
customerDetails=pd.read_csv("Customer_and_bank details_p1.csv")
customerPostalCode=pd.read_csv("Customer_Postal_Code_details.csv")
customerResponseData=pd.read_csv("Customer_Response_data_p1.csv")
customerCampaignDetails=pd.read_csv("Customer_campaign_details_p1.csv")
customerSocialEconomicData=pd.read_csv("Customer_social_economic_data_p1.csv")
regionCode=pd.read_csv("Region_code_master.csv")
stateCode=pd.read_csv("State_Master.csv")

In [7]:
customerCampaignDetails=customerCampaignDetails.rename(columns={'Customer_id':'C_id'})
customerDetails=pd.concat([customerDetails,customerCampaignDetails],axis=1)
customerDetails=pd.concat([customerDetails,customerPostalCode],axis=1)
del(customerDetails['customer_id'])
del(customerDetails['C_id'])
df1=pd.DataFrame(customerDetails['Region_Code'])
df2=[regionCode.loc[i][:] for i in [0,1,2,3,4]]
df2=pd.DataFrame(df2)
customerDetails=pd.merge(customerDetails,customerSocialEconomicData,how="left",on='Customer_id')
df1['Region']=df1['Region_Code'].replace(['1','2','3','4','5','Na'],['Central','East','South','West','North','Not_Specified'])
del[df1['Region_Code']]
df1['Customer_id']=customerDetails['Customer_id']
customerDetails=pd.merge(customerDetails,df1,how='inner',on='Customer_id')
 


In [15]:
bank.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 37084 entries, 0 to 37083
Data columns (total 27 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Customer_id     37084 non-null  int64  
 1   age             37084 non-null  int64  
 2   job             37084 non-null  object 
 3   marital         37084 non-null  object 
 4   education       37084 non-null  object 
 5   default         37084 non-null  object 
 6   housing         37084 non-null  object 
 7   loan            37084 non-null  object 
 8   Region_Code     37084 non-null  object 
 9   State_Code      37084 non-null  object 
 10  City_Code       37084 non-null  object 
 11  contact         37084 non-null  object 
 12  month           37084 non-null  object 
 13  day_of_week     37084 non-null  object 
 14  duration        37084 non-null  int64  
 15  campaign        37084 non-null  int64  
 16  pdays           37084 non-null  int64  
 17  previous        37084 non-null 

In [14]:
bank = pd.merge(customerDetails,customerResponseData,how = 'outer')
bank['contact'].unique()

array(['telephone', 'cellular'], dtype=object)

Customer_id  age          job  marital            education  default  \
0                1   56     services  married          high.school       no   
1                2   45     services  married             basic.9y  unknown   
2                3   59       admin.  married  professional.course       no   
3                4   41  blue-collar  married              unknown  unknown   
4                5   24   technician   single  professional.course       no   
...            ...  ...          ...      ...                  ...      ...   
37079        37080   73      retired  married  professional.course       no   
37080        37081   46  blue-collar  married  professional.course       no   
37081        37082   56      retired  married    university.degree       no   
37082        37083   44   technician  married  professional.course       no   
37083        37084   74      retired  married  professional.course       no   

      housing loan Region_Code State_Code  ... previous     poutcome  \
0          no  yes           3         S1  ...        0  nonexistent   
1          no   no           3         S1  ...        0  nonexistent   
2          no   no           4         S2  ...        0  nonexistent   
3          no   no           3         S3  ...        0  nonexistent   
4         yes   no           3         S3  ...        0  nonexistent   
...       ...  ...         ...        ...  ...      ...          ...   
37079     yes   no           2        S16  ...        0  nonexistent   
37080      no   no           2        S16  ...        0  nonexistent   
37081     yes   no           2        S16  ...        0  nonexistent   
37082      no   no           4        S17  ...        0  nonexistent   
37083     yes   no           1         S6  ...        1      failure   

      Postal Code emp.var.rate  cons.price.idx  cons.conf.idx  euribor3m  \
0           42420          1.1          93.994          -36.4      4.857   
1           42420          1.1          93.994          -36.4      4.857   
2           90036          1.1          93.994          -36.4      4.857   
3           33311          1.1          93.994          -36.4      4.857   
4           33311          1.1          93.994          -36.4      4.857   
...           ...          ...             ...            ...        ...   
37079       10009         -1.1          94.767          -50.8      1.028   
37080       10011         -1.1          94.767          -50.8      1.028   
37081       10009         -1.1          94.767          -50.8      1.028   
37082       85254         -1.1          94.767          -50.8      1.028   
37083       79109         -1.1          94.767          -50.8      1.028   

       nr.employed   Region    y  
0           5191.0    South   no  
1           5191.0    South   no  
2           5191.0     West   no  
3           5191.0    South   no  
4           5191.0    South   no  
...            ...      ...  ...  
37079       4963.6     East  yes  
37080       4963.6     East   no  
37081       4963.6     East   no  
37082       4963.6     West  yes  
37083       4963.6  Central   no  

[37084 rows x 27 columns]

In [7]:
bank['contact'].value_counts()/len(bank['contact'])*100

cellular     63.428972
telephone    36.571028
Name: contact, dtype: float64

Note: 63.42 % of the clients were approached through cellular contact and 36.57% of the clients were approached through telephonic communication

In [8]:
print(bank[bank['y']=='yes']['contact'].value_counts()/len(bank[bank['y']=='yes']['contact'])*100)

cellular     82.984791
telephone    17.015209
Name: contact, dtype: float64


Of all the people who have subscribed 82.9% clients were approached through cellular contact and only 17.01% people who were approached through telephone have subscribed. Thus making Celullar Mode of contact the more efficient one

# 3: Analysis on attempts made to turn a person into successful depositor

In [1]:
import pandas as pd
from pandas import Series,DataFrame
import numpy as np

In [2]:
campaign_details=pd.read_csv('Customer_campaign_details_p1.csv')
campaign_details.head()

Customer_id    contact month day_of_week  duration  campaign  pdays  \
0            1  telephone   may         mon       307         1    999   
1            2  telephone   may         mon       198         1    999   
2            3  telephone   may         mon       139         1    999   
3            4  telephone   may         mon       217         1    999   
4            5  telephone   may         mon       380         1    999   

   previous     poutcome  
0         0  nonexistent  
1         0  nonexistent  
2         0  nonexistent  
3         0  nonexistent  
4         0  nonexistent

In [3]:
response=pd.read_csv('Customer_Response_data_p1.csv')
response.head()

Customer_id   y
0            1  no
1            2  no
2            3  no
3            4  no
4            5  no

In [4]:
response_merged=pd.merge(campaign_details,response,on='Customer_id',how='inner')
response_merged.head(10)

Customer_id    contact month day_of_week  duration  campaign  pdays  \
0            1  telephone   may         mon       307         1    999   
1            2  telephone   may         mon       198         1    999   
2            3  telephone   may         mon       139         1    999   
3            4  telephone   may         mon       217         1    999   
4            5  telephone   may         mon       380         1    999   
5            6  telephone   may         mon        50         1    999   
6            7  telephone   may         mon        55         1    999   
7            8  telephone   may         mon       222         1    999   
8            9  telephone   may         mon       137         1    999   
9           10  telephone   may         mon       293         1    999   

   previous     poutcome   y  
0         0  nonexistent  no  
1         0  nonexistent  no  
2         0  nonexistent  no  
3         0  nonexistent  no  
4         0  nonexistent  no  
5         0  nonexistent  no  
6         0  nonexistent  no  
7         0  nonexistent  no  
8         0  nonexistent  no  
9         0  nonexistent  no

### Analysis on the attempts

In [11]:
max_attempts=response_merged.query("y=='yes'")['campaign'].max()
min_attempts=response_merged.query("y=='yes'")['campaign'].min()
avg=response_merged.query("y=='yes'")['campaign'].mean()
print('Maximum attempts:',max_attempts)
print('Minimum attempts:',min_attempts)
print('Average of attempts:',int(avg))

Maximum attempts: 23
Minimum attempts: 1
Average of attempts: 2


# 4.Personal data analysis on marital status, existing loans, education, profession etc. and its impact on the campaign’s success.

In [1]:
import numpy as np
import pandas as pd
from pandas import DataFrame

#### Displaying the Tables

In [2]:
df0=pd.read_csv('Customer_campaign_details_p1.csv')
df0
df0.head()

Customer_id    contact month day_of_week  duration  campaign  pdays  \
0            1  telephone   may         mon       307         1    999   
1            2  telephone   may         mon       198         1    999   
2            3  telephone   may         mon       139         1    999   
3            4  telephone   may         mon       217         1    999   
4            5  telephone   may         mon       380         1    999   

   previous     poutcome  
0         0  nonexistent  
1         0  nonexistent  
2         0  nonexistent  
3         0  nonexistent  
4         0  nonexistent

In [3]:
df1 = pd.read_csv('Customer_and_bank details_p1.csv')
df1
df1.head()


Customer_id  age          job  marital            education  default  \
0            1   56     services  married          high.school       no   
1            2   45     services  married             basic.9y  unknown   
2            3   59       admin.  married  professional.course       no   
3            4   41  blue-collar  married              unknown  unknown   
4            5   24   technician   single  professional.course       no   

  housing loan Region_Code State_Code City_Code  
0      no  yes           3         S1        C1  
1      no   no           3         S1        C1  
2      no   no           4         S2        C2  
3      no   no           3         S3        C3  
4     yes   no           3         S3        C3

In [4]:
df2 = pd.read_csv('Customer_Response_data_p1.csv')
df2
df2.head()

Customer_id   y
0            1  no
1            2  no
2            3  no
3            4  no
4            5  no

#### merging Customer_and_bank details_p1.csv and Customer_Response_data_p1.csv(tables)


In [5]:
grouph=pd.merge(df1,df2,on='Customer_id',how='inner')
grouph.head()

Customer_id  age          job  marital            education  default  \
0            1   56     services  married          high.school       no   
1            2   45     services  married             basic.9y  unknown   
2            3   59       admin.  married  professional.course       no   
3            4   41  blue-collar  married              unknown  unknown   
4            5   24   technician   single  professional.course       no   

  housing loan Region_Code State_Code City_Code   y  
0      no  yes           3         S1        C1  no  
1      no   no           3         S1        C1  no  
2      no   no           4         S2        C2  no  
3      no   no           3         S3        C3  no  
4     yes   no           3         S3        C3  no

#### analysis on martial status based on responses

In [6]:
grouph1=grouph.groupby('marital')['y'].value_counts()
grouph1

marital   y  
divorced  no      3699
          yes      427
married   no     20187
          yes     2292
single    no      8930
          yes     1477
unknown   no        60
          yes       12
Name: y, dtype: int64

#### analysis on loan based on responses

In [7]:
grouph2=grouph.groupby('loan')['y'].value_counts()
grouph2

loan     y  
no       no     27062
         yes     3499
unknown  no       791
         yes       91
yes      no      5023
         yes      618
Name: y, dtype: int64

#### analysis on education based on responses

In [8]:
grouph3=grouph.groupby('education')['y'].value_counts()
grouph3

education            y  
basic.4y             no     3374
                     yes     391
basic.6y             no     1903
                     yes     171
basic.9y             no     5001
                     yes     420
high.school          no     7597
                     yes     945
illiterate           no       14
                     yes       2
professional.course  no     4205
                     yes     541
university.degree    no     9464
                     yes    1507
unknown              no     1318
                     yes     231
Name: y, dtype: int64

#### analysis on profession based on responses

In [9]:
grouph4=grouph.groupby('job')['y'].value_counts()
grouph4

job            y  
admin.         no     8174
               yes    1246
blue-collar    no     7738
               yes     576
entrepreneur   no     1198
               yes     111
housemaid      no      853
               yes      93
management     no     2345
               yes     292
retired        no     1146
               yes     395
self-employed  no     1145
               yes     131
services       no     3249
               yes     298
student        no      527
               yes     240
technician     no     5439
               yes     657
unemployed     no      791
               yes     134
unknown        no      271
               yes      35
Name: y, dtype: int64

#### analysis on housing based on responses

In [10]:
grouph5=grouph.groupby('housing')['y'].value_counts()
grouph5

housing  y  
no       no     14931
         yes     1838
unknown  no       791
         yes       91
yes      no     17154
         yes     2279
Name: y, dtype: int64

#### analysis on age based on responses

In [15]:
grouph6=grouph.groupby('age')['y'].value_counts()
grouph6

age  y  
17   no      3
     yes     2
18   no     15
     yes     9
19   no     21
            ..
92   yes     3
     no      1
94   no      1
95   no      1
98   yes     2
Name: y, Length: 149, dtype: int64

#### merging Customer_and_bank details_p1.csv and Customer_Campaign_data_p1.csv(tables)

In [11]:
group=pd.merge(df0,df1,on='Customer_id',how='inner')
group
group.head()

Customer_id    contact month day_of_week  duration  campaign  pdays  \
0            1  telephone   may         mon       307         1    999   
1            2  telephone   may         mon       198         1    999   
2            3  telephone   may         mon       139         1    999   
3            4  telephone   may         mon       217         1    999   
4            5  telephone   may         mon       380         1    999   

   previous     poutcome  age          job  marital            education  \
0         0  nonexistent   56     services  married          high.school   
1         0  nonexistent   45     services  married             basic.9y   
2         0  nonexistent   59       admin.  married  professional.course   
3         0  nonexistent   41  blue-collar  married              unknown   
4         0  nonexistent   24   technician   single  professional.course   

   default housing loan Region_Code State_Code City_Code  
0       no      no  yes           3         S1        C1  
1  unknown      no   no           3         S1        C1  
2       no      no   no           4         S2        C2  
3  unknown      no   no           3         S3        C3  
4       no     yes   no           3         S3        C3

#### analysis on martial status based on poutcome

In [25]:
marital_success=group.groupby('marital')['poutcome'].value_counts()
marital_success

marital   poutcome   
divorced  nonexistent     3562
          failure          452
          success          112
married   nonexistent    19674
          failure         2146
          success          659
single    nonexistent     8729
          failure         1215
          success          463
unknown   nonexistent       58
          failure            9
          success            5
Name: poutcome, dtype: int64


#### calculating percentage on martial status analysis

In [44]:
print('The percentage turnout of maritalstatus')

The percentage turnout of maritalstatus


In [45]:
percent = group['marital'].value_counts()/len(group['marital'])*100
percent


married     60.616438
single      28.063316
divorced    11.126092
unknown      0.194154
Name: marital, dtype: float64

#### analysis on existing loans based on poutcome

In [47]:
loan_success=group.groupby('loan')['poutcome'].value_counts()
loan_success

loan     poutcome   
no       nonexistent    26389
         failure         3141
         success         1031
unknown  nonexistent      756
         failure          101
         success           25
yes      nonexistent     4878
         failure          580
         success          183
Name: poutcome, dtype: int64

#### calculating percentage on existing analysis

In [46]:
print('The percentage turnout of loan')

The percentage turnout of loan


In [42]:
percent1 = group['loan'].value_counts()/len(group['loan'])*100
percent1

no         82.410204
yes        15.211412
unknown     2.378384
Name: loan, dtype: float64

#### analysis on education based on poutcome

In [49]:
education_success=group.groupby('education')['poutcome'].value_counts()
education_success

education            poutcome   
basic.4y             nonexistent    3332
                     failure         316
                     success         117
basic.6y             nonexistent    1838
                     failure         200
                     success          36
basic.9y             nonexistent    4764
                     failure         563
                     success          94
high.school          nonexistent    7278
                     failure         996
                     success         268
illiterate           nonexistent      14
                     failure           1
                     success           1
professional.course  nonexistent    4116
                     failure         463
                     success         167
university.degree    nonexistent    9375
                     failure        1118
                     success         478
unknown              nonexistent    1306
                     failure         165
                     suc

#### calculating percentage on education analysis

In [50]:
print('The percentage turnout of education')

The percentage turnout of education


In [51]:
percent2 = group['education'].value_counts()/len(group['education'])*100
percent2

university.degree      29.584187
high.school            23.034193
basic.9y               14.618164
professional.course    12.797972
basic.4y               10.152626
basic.6y                5.592708
unknown                 4.177004
illiterate              0.043145
Name: education, dtype: float64

#### analysis on profession based on poutcome

In [13]:
job_success=group.groupby('job')['poutcome'].value_counts()
job_success

job            poutcome   
admin.         nonexistent    8037
               failure         987
               success         396
blue-collar    nonexistent    7412
               failure         800
               success         102
entrepreneur   nonexistent    1147
               failure         140
               success          22
housemaid      nonexistent     847
               failure          64
               success          35
management     nonexistent    2257
               failure         292
               success          88
retired        nonexistent    1195
               failure         203
               success         143
self-employed  nonexistent    1119
               failure         132
               success          25
services       nonexistent    3088
               failure         397
               success          62
student        nonexistent     519
               failure         138
               success         110
technician     nonexistent  

#### calculating percentage on profession analysis

In [14]:
percent4= group['job'].value_counts()/len(group['job'])*100
percent4

admin.           25.401791
blue-collar      22.419372
technician       16.438356
services          9.564772
management        7.110883
retired           4.155431
entrepreneur      3.529824
self-employed     3.440837
housemaid         2.550965
unemployed        2.494337
student           2.068277
unknown           0.825154
Name: job, dtype: float64

#### analysis on housing based on poutcome

In [53]:
housing_success=group.groupby('housing')['poutcome'].value_counts()
housing_success

housing  poutcome   
no       nonexistent    14662
         failure         1594
         success          513
unknown  nonexistent      756
         failure          101
         success           25
yes      nonexistent    16605
         failure         2127
         success          701
Name: poutcome, dtype: int64

#### calculating percentage on housing analysis

In [54]:
print('The percentage turnout of housing')

The percentage turnout of housing


In [55]:
percent3 = group['housing'].value_counts()/len(group['housing'])*100
percent3

yes        52.402653
no         45.218962
unknown     2.378384
Name: housing, dtype: float64

# 5.Socio-economical analysis of the customers

In [3]:
import pandas as pd
import numpy as np
from pandas import Series,DataFrame

### Reading the Dataset related to Socio-Economic details of the customers

In [4]:
SocioEconomicData=pd.read_csv('Customer_social_economic_data_p1.csv') 

#### Counting the number of employees with Negative and Positive variation rate

In [5]:
Neg_Var_rate=SocioEconomicData[SocioEconomicData['emp.var.rate']<0].shape[0]
Pos_var_rate=SocioEconomicData[SocioEconomicData['emp.var.rate']>0].shape[0]

In [18]:
print("Among the total of ",Neg_Var_rate+Pos_var_rate,' employees')
print(f'The number of employess with negative variation rate is {Neg_Var_rate} which is {round(Neg_Var_rate*100/len(SocioEconomicData),2)} %\nThe number of employees with positive variation rate is {Pos_var_rate}  which is {round(Pos_var_rate*100/len(SocioEconomicData),2)} %')


Among the total of  37084  employees
The number of employess with negative variation rate is 15486 which is 41.76 %
The number of employees with positive variation rate is 21598  which is 58.24 %


In [16]:
pd.options.display.float_format='{:.1f}'.format
SocioEconomicData.describe()

emp.var.rate  cons.price.idx  cons.conf.idx  euribor3m  nr.employed
count       37084.0         37084.0        37084.0    37084.0      37084.0
mean            0.1            93.6          -40.5        3.6       5167.1
std             1.6             0.6            4.6        1.7         72.2
min            -3.4            92.2          -50.8        0.6       4963.6
25%            -1.8            93.1          -42.7        1.3       5099.1
50%             1.1            93.7          -41.8        4.9       5191.0
75%             1.4            94.0          -36.4        5.0       5228.1
max             1.4            94.8          -26.9        5.0       5228.1

#### The average consumer price index is 93.6 where the minimum is at 92.2 and maximum at 94.8
#### The average consumer confidence index is -40.5 where the minimum is at -50.8 and maximum at -26.9
#### The maximum interest for a customer is 5% and the minimum is 0.6%

# 6.Demographic analysis of the marketing campaign using the master files.

###### Based on the region_names from the Region_code_master csv file:

In [89]:
bank['Region'].unique()

array(['South', 'West', 'Not_Specified', 'Central', 'East'], dtype=object)

## What is the count of customers from each location(region):

##### Central_region:

In [88]:
Region_central = bank[bank['Region'] == 'Central']
round(Region_central['Customer_id'].count(),2)

8590

##### South_region:

In [94]:
Region_South = bank[bank['Region'] == 'South']
round(Region_South['Customer_id'].count(),2)

5992

##### West_region:

In [95]:
Region_west = bank[bank['Region'] == 'West']
round(Region_west['Customer_id'].count(),2)

11884

##### North_region:

In [90]:
Region_north = bank[bank['Region'] == 'North']
round(Region_north['Customer_id'].count(),2)

0

Note: As you can see there are no customers from North Region. Thus, no need of analysis of this region.

##### East_region:

In [91]:
Region_east = bank[bank['Region'] == 'East']
round(Region_east['Customer_id'].count(),2)

10558

##### Not_Specified_region:

In [92]:
Region_ns = bank[bank['Region'] == 'Not_Specified']
round(Region_ns['Customer_id'].count(),2)

60

# What Is The Average Age in all the Regions:

#### Central_Region:

In [10]:
Region_central = bank[bank['Region'] == 'Central']
round(Region_central['age'].mean(),2)

40.14

#### East_Region:

In [11]:
Region_east = bank[bank['Region'] == 'East']
round(Region_east['age'].mean(),2)

40.03

#### Region not Specified:

In [12]:
Region_Not_specified = bank[bank['Region'] == 'Not_Specified']
Region_Not_specified['age'].mean()

39.5

#### South_Region:

In [13]:
Region_South = bank[bank['Region'] == 'South']
round(Region_South['age'].mean(),2)

40.08

#### West_Region:

In [14]:
Region_West = bank[bank['Region'] == 'West']
round(Region_West['age'].mean(),2)

39.97

# What Is The Percentage distribution Of education level In all the Regions?

In [15]:
print("Percentage distribution of education level of people in Central Region:\n")
print(bank[bank['Region']=='Central']['education'].value_counts()/len(bank[bank['Region']=='Central']['education'])*100)

Percentage distribution of education level of people in Central Region:

university.degree      30.128056
high.school            22.619325
basic.9y               14.388824
professional.course    12.805588
basic.4y               10.360885
basic.6y                5.320140
unknown                 4.318976
illiterate              0.058207
Name: education, dtype: float64


In [16]:
print("Percentage distribution of education level of people in West Region:\n")
print(bank[bank['Region']=='West']['education'].value_counts()/len(bank[bank['Region']=='West']['education'])*100)

Percentage distribution of education level of people in West Region:

university.degree      29.510266
high.school            23.056210
basic.9y               14.835072
professional.course    12.722989
basic.4y               10.114440
basic.6y                5.503198
unknown                 4.224167
illiterate              0.033659
Name: education, dtype: float64


In [37]:
print("Percentage distribution of education level of people in East Region:\n")
print(bank[bank['Region']=='East']['education'].value_counts()/len(bank[bank['Region']=='East']['education'])*100)

Percentage distribution of education level of people in East Region:

university.degree      29.143777
high.school            23.167267
basic.9y               14.728168
professional.course    12.947528
basic.4y                9.859822
basic.6y                5.843910
unknown                 4.262171
illiterate              0.047357
Name: education, dtype: float64


In [38]:
print("Percentage distribution of education level of people in South Region:\n")
print(bank[bank['Region']=='South']['education'].value_counts()/len(bank[bank['Region']=='South']['education'])*100)

Percentage distribution of education level of people in South Region:

university.degree      29.672897
high.school            23.431242
basic.9y               14.352470
professional.course    12.733645
basic.4y               10.397196
basic.6y                5.690921
unknown                 3.688251
illiterate              0.033378
Name: education, dtype: float64


In [39]:
print("Percentage distribution of education level of people in Not_specified Region:\n")
print(bank[bank['Region']=='Not_Specified']['education'].value_counts()/len(bank[bank['Region']=='Not_Specified']['education'])*100)

Percentage distribution of education level of people in Not_specified Region:

university.degree      35.000000
basic.4y               15.000000
high.school            15.000000
basic.9y               11.666667
basic.6y                8.333333
unknown                 8.333333
professional.course     6.666667
Name: education, dtype: float64


# What Is The Percentage distribution Of jobs In all the Regions?

In [65]:
print("Percentage distribution of jobs of people in Central Region:\n")
print(bank[bank['Region']=='Central']['job'].value_counts()/len(bank[bank['Region']=='Central']['job'])*100)

Percentage distribution of jobs of people in Central Region:

admin.           25.715949
blue-collar      21.781141
technician       16.903376
services          9.045402
management        7.427241
retired           4.516880
self-employed     3.410943
entrepreneur      3.294529
housemaid         2.828871
unemployed        2.409779
student           1.757858
unknown           0.908033
Name: job, dtype: float64


In [66]:
print("Percentage distribution of jobs of people in West Region:\n")
print(bank[bank['Region']=='West']['job'].value_counts()/len(bank[bank['Region']=='West']['job'])*100)

Percentage distribution of jobs of people in West Region:

admin.           25.412319
blue-collar      22.214743
technician       16.189835
services          9.634803
management        6.933692
retired           4.081117
entrepreneur      3.862336
self-employed     3.500505
housemaid         2.532817
unemployed        2.457085
student           2.381353
unknown           0.799394
Name: job, dtype: float64


In [67]:
print("Percentage distribution of jobs of people in South Region:\n")
print(bank[bank['Region']=='South']['job'].value_counts()/len(bank[bank['Region']=='South']['job'])*100)

Percentage distribution of jobs of people in South Region:

admin.           25.033378
blue-collar      23.230975
technician       15.987984
services          9.963284
management        6.859146
retired           4.155541
entrepreneur      3.421228
self-employed     3.421228
unemployed        2.503338
housemaid         2.419893
student           2.303071
unknown           0.700935
Name: job, dtype: float64


In [68]:
print("Percentage distribution of jobs of people in Not_Specified Region:\n")
print(bank[bank['Region']=='Not_Specified']['job'].value_counts()/len(bank[bank['Region']=='Not_Specified']['job'])*100)

Percentage distribution of jobs of people in Not_Specified Region:

blue-collar      28.333333
admin.           28.333333
technician       15.000000
services          8.333333
housemaid         6.666667
entrepreneur      3.333333
self-employed     3.333333
management        3.333333
retired           3.333333
Name: job, dtype: float64


In [69]:
print("Percentage distribution of jobs of people in East Region:\n")
print(bank[bank['Region']=='East']['job'].value_counts()/len(bank[bank['Region']=='East']['job'])*100)

Percentage distribution of jobs of people in East Region:

admin.           25.326766
blue-collar      22.674749
technician       16.603523
services          9.689335
management        7.217276
retired           3.949612
entrepreneur      3.409737
self-employed     3.409737
unemployed        2.614131
housemaid         2.396287
student           1.846941
unknown           0.861906
Name: job, dtype: float64


# What Is The Percentage distribution Of ages In all the Regions?

In [106]:
print("Percentage distribution of ages of people in Central Region:\n")
df= pd.DataFrame(bank[bank['Region']=='Central']['age'].value_counts()/len(bank[bank['Region']=='Central']['age'])*100)
df


Percentage distribution of ages of people in Central Region:



age
31  4.959255
32  4.784633
35  4.412107
36  4.353900
34  4.225844
..       ...
91  0.011641
94  0.011641
84  0.011641
89  0.011641
95  0.011641

[75 rows x 1 columns]

In [91]:
print("Percentage distribution of ages of people in West Region:\n")
print(bank[bank['Region']=='West']['age'].value_counts()/len(bank[bank['Region']=='West']['age'])*100)

Percentage distribution of ages of people in West Region:

36    4.720633
33    4.661730
31    4.510266
32    4.434534
30    4.358802
        ...   
84    0.016829
92    0.008415
98    0.008415
17    0.008415
91    0.008415
Name: age, Length: 74, dtype: float64


In [92]:
print("Percentage distribution of ages of people in South Region:\n")
print(bank[bank['Region']=='South']['age'].value_counts()/len(bank[bank['Region']=='South']['age'])*100)

Percentage distribution of ages of people in South Region:

31    4.606142
34    4.539386
33    4.439252
32    4.172230
36    4.172230
        ...   
82    0.016689
89    0.016689
85    0.016689
84    0.016689
83    0.016689
Name: age, Length: 72, dtype: float64


In [107]:
print("Percentage distribution of ages of people in Not_Specified Region:\n")
print(bank[bank['Region']=='Not_Specified']['age'].value_counts()/len(bank[bank['Region']=='Not_Specified']['age'])*100)

Percentage distribution of ages of people in Not_Specified Region:

38    10.000000
33     6.666667
27     6.666667
42     6.666667
40     5.000000
31     5.000000
29     5.000000
47     3.333333
46     3.333333
43     3.333333
41     3.333333
30     3.333333
37     3.333333
35     3.333333
34     3.333333
32     3.333333
55     3.333333
58     1.666667
54     1.666667
59     1.666667
57     1.666667
56     1.666667
26     1.666667
53     1.666667
51     1.666667
48     1.666667
45     1.666667
39     1.666667
28     1.666667
60     1.666667
Name: age, dtype: float64


In [108]:
print("Percentage distribution of ages of people in East Region:\n")
print(bank[bank['Region']=='East']['age'].value_counts()/len(bank[bank['Region']=='East']['age'])*100)

Percentage distribution of ages of people in East Region:

31    4.925175
33    4.593673
34    4.375829
32    4.300057
35    4.252699
        ...   
74    0.018943
17    0.018943
81    0.018943
80    0.009471
79    0.009471
Name: age, Length: 72, dtype: float64


# What Is The Percentage distribution Of marital status In all the Regions?

In [136]:
print("Percentage distribution of marital status of people in Central Region:\n")
print(bank[bank['Region']=='Central']['marital'].value_counts()/len(bank[bank['Region']=='Central']['marital'])*100)

Percentage distribution of marital status of people in Central Region:

married     60.838184
single      27.916182
divorced    11.047730
unknown      0.197905
Name: marital, dtype: float64


In [137]:
print("Percentage distribution of marital status of people in West Region:\n")
print(bank[bank['Region']=='West']['marital'].value_counts()/len(bank[bank['Region']=='West']['marital'])*100)

Percentage distribution of marital status of people in West Region:

married     59.971390
single      28.660384
divorced    11.191518
unknown      0.176708
Name: marital, dtype: float64


In [138]:
print("Percentage distribution of marital status of people in East Region:\n")
print(bank[bank['Region']=='East']['marital'].value_counts()/len(bank[bank['Region']=='East']['marital'])*100)

Percentage distribution of marital status of people in East Region:

married     60.996401
single      27.675696
divorced    11.110059
unknown      0.217844
Name: marital, dtype: float64


In [139]:
print("Percentage distribution of marital status of people in Not_Specidied Region:\n")
print(bank[bank['Region']=='Not_Specified']['marital'].value_counts()/len(bank[bank['Region']=='Not_Specified']['marital'])*100)

Percentage distribution of marital status of people in Not_Specidied Region:

married     53.333333
single      28.333333
divorced    18.333333
Name: marital, dtype: float64


In [140]:
print("Percentage distribution of marital status of people in South Region:\n")
print(bank[bank['Region']=='South']['marital'].value_counts()/len(bank[bank['Region']=='South']['marital'])*100)

Percentage distribution of marital status of people in South Region:

married     60.981308
single      27.770360
divorced    11.064753
unknown      0.183578
Name: marital, dtype: float64
